src/client/state/RoomsInput.js

```diff
@@ -200,6 +201,42 @@ class RoomsInput extends EventEmitter {
     return this.roomIdToInput.get(roomId)?.isSending || false;
   }
 
+  // Apply formatting to a plain text message
+  //
+  // This includes inserting any custom emoji that might be relevant, and (only if the
+  // user has enabled it in their settings) formatting the message using markdown.
+  formatAndEmojifyText(text) {
+    // Check to see if there are any :shortcode-style-tags: in the message
+    const shortcodes = Array.from(text.matchAll(/\B:([\w-]+):\B/g)).map((m) => m[1]);
+    // Find the corresponding emoji
+    const possibleEmoji = getUserEmoji(this.matrixClient)
+      .filter((e) => shortcodes.indexOf(e.shortcode) !== -1);
+    // But we'll put the emoji in after, so that they don't get escaped by the markdown
+    // formatter
+
+    // Now we apply markdown formatting
+    let formattedText;
+    if (settings.isMarkdown) {
+      formattedText = getFormattedBody(text);
+    }
+
+    // If there were any emojis, now we substitute :shortcodes: for the <img/> tag
+    for (let i = 0; i < possibleEmoji.length; i += 1) {
+      const emoji = possibleEmoji[i];
+      const tag = `<img data-mx-emoticon="" src="${
+        emoji.mxc
+      }" alt=":${
+        emoji.shortcode
+      }:" title=":${
+        emoji.shortcode
+      }:" height="32" />`;
+
+      formattedText = formattedText.replaceAll(`:${emoji.shortcode}:`, tag);
```

Comment:
it gives me an error that can't read `replaceAll` of undefined when sending with markdown disabled. as formattedText is assigned only if markdown is enabled. see line: 219